crates/stages/src/stages/headers.rs

```diff
@@ -130,7 +122,14 @@ impl<DB: Database, D: HeaderDownloader, C: Consensus, H: HeadersClient, S: Statu
         // Write total difficulty values after all headers have been inserted
         self.write_td::<DB>(db, &head)?;
 
-        Ok(ExecOutput { stage_progress: current_progress, reached_tip: true, done: true })
+        let stage_progress = current_progress.max(
+            db.cursor::<tables::CanonicalHeaders>()?
+                .last()?
+                .map(|(num, _)| num)
+                .unwrap_or_default(),
+        );
```

Comment:
nice. @onbjerg pls confirm this is correct. is this supposed to be `.last()?`

Reply:
yes but only because we are committing inside of the stage currently, so this code path will only be hit once the entire range is synced (and there are no gaps)

Reply:
let's add this exact note in a doc comment above this line